### PR TITLE
PLAT-954 Enable deactivation of BufferedEmail

### PR DESCRIPTION
**Pull request by daniel-klose**

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/AGBufferedEmailLog.sqml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+PACKAGE com.softicar.platform.core.module.email.buffer
+
+IMPORT com.softicar.platform.core.module.transaction.AGTransaction
+
+TABLE AGBufferedEmailLog 'Core.BufferedEmailLog' {
+	
+	PK AGBufferedEmail bufferedEmail
+	PK AGTransaction transaction
+	Boolean active = NULL [NULLABLE]
+}
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/AGBufferedEmailLog.sqml:10]: add missing EOL

File: platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public void sendForAllServers() {
 
 			AGBufferedEmail.TABLE//
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java:43]: who has written this code? shouldn't this start with AGBufferedEmailServer.TABLE.createSelect()?????

File: platform-core-module/src/test/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSenderTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Test;
+
+public class BufferedEmailSenderTest extends AbstractCoreTest {
+
+	private final AGServer testServer;
+
+	public BufferedEmailSenderTest() {
+
+		this.testServer = insertServer("Server", "1.1.1.1", 25, "domain", "username", "password");
+		System.setProperty(EmailSystemProperties.SENDING_ENABLED.getPropertyName().toString(), "false");
```
Comment (oliver-richers) [inline platform-core-module/src/test/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSenderTest.java:16]: - modifying these global properties is quite ugly and dangerous (can lead to hard to find bugs in the tests)
- but I also think there is hardly a way around it, without an abstraction-layer

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java (hunk 2 above)
Comment (daniel-klose) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java:43]: I think this was requested.
Only check servers where BufferedEmails exists.
Dunno when and why

File: platform-core-module/src/test/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSenderTest.java (hunk 3 above)
Comment (oliver-richers) [reply platform-core-module/src/test/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSenderTest.java:16]: suggestion: add a TODO with reference to a story that will fix this through an abstraction layer

Comment (daniel-klose) [reply platform-core-module/src/test/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSenderTest.java:16]: As I said, this doesnt work.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java (hunk 2 above)
Comment (daniel-klose) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java:43]: I can rewrite it a bit

File: platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java [hunk 4]
```diff
@@ -40,14 +40,12 @@ public Sender() {
 
 		public void sendForAllServers() {
 
-			AGBufferedEmail.TABLE//
+			AGServer.TABLE//
 				.createSelect()
-				.where(AGBufferedEmail.SENT_AT.isNull())
-				.groupBy(AGBufferedEmail.EMAIL_SERVER)
-				.join(AGBufferedEmail.EMAIL_SERVER)
 				.where(AGServer.ACTIVE)
```
Comment (oliver-richers) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/email/buffer/BufferedEmailSendProgram.java:45]: uh, we may not remove the groupBy :)

**Review by oliver-richers — CHANGES_REQUESTED** [review]

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**